### PR TITLE
[WFMP-161] Allow the feature pack location to be overridden for the s…

### PR DESCRIPTION
**Pull request by jamezp**
…tart and run goals.

https://issues.redhat.com/browse/WFMP-161

Upstream #237

### CODE REVIEW

File: plugin/src/main/java/org/wildfly/plugin/server/RunMojo.java [hunk 1]
```diff
@@ -360,7 +371,7 @@ private Path provisionIfRequired(final Path installDir) throws MojoFailureExcept
         try {
             if (!Files.exists(installDir)) {
                 getLog().info("Provisioning default WildFly server in " + installDir);
```
Comment (jfdenise) [inline plugin/src/main/java/org/wildfly/plugin/server/RunMojo.java:373]: Perhaps we could change this sentence to: "Provisioning default server in"

File: core/src/main/java/org/wildfly/plugin/core/GalleonUtils.java [hunk 2]
```diff
@@ -247,9 +248,9 @@ private static String getMavenCoords(FeaturePack fp) {
         return builder.toString();
     }
 
-    private static String getWildFlyFeaturePackLocation(String version) {
+    private static String getWildFlyFeaturePackLocation(String featurePackLocation, String version) {
         StringBuilder fplBuilder = new StringBuilder();
-        fplBuilder.append(WILDFLY_DEFAULT_FEATURE_PACK_LOCATION);
+        fplBuilder.append(featurePackLocation);
```
Comment (jfdenise) [inline core/src/main/java/org/wildfly/plugin/core/GalleonUtils.java:253]: We could throw an exception if featurepackLocation is null.

File: plugin/src/main/java/org/wildfly/plugin/server/RunMojo.java (hunk 1 above)
Comment (jamezp) [reply plugin/src/main/java/org/wildfly/plugin/server/RunMojo.java:373]: +1 that makes sense.

File: core/src/main/java/org/wildfly/plugin/core/GalleonUtils.java (hunk 2 above)
Comment (jamezp) [reply core/src/main/java/org/wildfly/plugin/core/GalleonUtils.java:253]: Good point. When I tested manually it did error out, but being more explicit probably makes more sense. I wonder if we should validate the pattern too.

**Pull request merged**